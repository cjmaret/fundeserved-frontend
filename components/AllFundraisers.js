import { useQuery } from '@apollo/client';
import gql from 'graphql-tag';
import Link from 'next/link';
import { featuredCards } from '../array-data/featured-data';
import Fundraiser from './Fundraiser';
import {
  CreateFundraiserButton,
  FundraisersGrid,
  FundraisersLead,
  FundraisersSubtitle,
  FundraisersTitle,
} from './styles/styledFundraisers';

export const ALL_FUNDRAISERS_QUERY = gql`
  query ALL_FUNDRAISERS_QUERY {
    allFundraisers {
      id
      name
      amount
      description
      goal
      dateCreated
      photo {
        image {
          publicUrlTransformed
        }
      }
    }
  }
`;

export default function AllFundraisers() {
  const { data, error, loading } = useQuery(ALL_FUNDRAISERS_QUERY);

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error: {error.message}</p>;
  return (
    <>
      <FundraisersLead>
        <FundraisersTitle>Browse fundraisers</FundraisersTitle>
        <FundraisersSubtitle>
          There's always something stupid to give your money to.
        </FundraisersSubtitle>
        <Link href="/create-fundraiser">
          <CreateFundraiserButton>Start a Fundeserved</CreateFundraiserButton>
        </Link>
      </FundraisersLead>
      <FundraisersGrid>
        {data.allFundraisers.map((fundraiser) => (
          <Fundraiser key={fundraiser.id} fundraiser={fundraiser} />
        ))}
      </FundraisersGrid>
    </>
  );
}
