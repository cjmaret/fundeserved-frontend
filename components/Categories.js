import {
  Card,
  CardImageWrapper,
  CardTitle,
  CategoriesComponent,
  CategoriesParagraph,
  CategoriesTitle,
  GridGroup,
  FilmCover,
  LinkCover,
} from './styles/styledCategories';
import { categoriesCards } from '../array-data/categories-data';
import Image from 'next/image';

export default function Categories() {
  return (
    <CategoriesComponent>
      <CategoriesTitle>Choose your fundeserved category</CategoriesTitle>
      <CategoriesParagraph>Which fraud interests you most?</CategoriesParagraph>

      <GridGroup>
        {categoriesCards.map((card) => (
          <Card key={card.id}>
            <LinkCover href="" />
            <CardTitle>{card.title}</CardTitle>
            <FilmCover />
            <CardImageWrapper>
              <Image
                src={card.image}
                alt=""
                layout="fill"
                objectFit="cover"
              />
            </CardImageWrapper>
          </Card>
        ))}
      </GridGroup>
    </CategoriesComponent>
  );
}
