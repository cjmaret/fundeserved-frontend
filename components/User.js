import { gql, useQuery } from '@apollo/client';

export const CURRENT_USER_QUERY = gql`
  query CURRENT_USER_QUERY {
    authenticatedItem {
      ... on User {
        id
        name
        email
        createdOn
        avatar {
          publicUrlTransformed
        }
        fundraisers(sortBy: dateCreated_DESC) {
          id
          name
          description
          amount
          goal
          dateCreated
          photo {
            image {
              publicUrlTransformed
            }
          }
        }
        donations(sortBy: dateCreated_DESC) {
          id
          total
          dateCreated
          fundraiser {
            id
            name
            description
            photo {
              image {
                publicUrlTransformed
              }
            }
          }
        }
      }
    }
  }
`;

export function useUser() {
  const { data } = useQuery(CURRENT_USER_QUERY);
  return data?.authenticatedItem;
}
