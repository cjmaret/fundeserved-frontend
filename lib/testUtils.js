import casual from 'casual';
import { PAGINATION_COUNT_QUERY } from '../components/Pagination';

casual.seed(777);

const fakeFundraiser = () => ({
  id: 'abc123',
  name: 'test-fundraiser',
  description: 'fundraiser description',
  status: 'ACTIVE',
  amount: 500,
  goal: 600,
  dateCreated: '2022-08-25T16:00:00.000Z',
  user: {
    id: '123',
  },
  photo: {
    id: 'abc123',
    altText: 'test-fundraiser',
    image: {
      publicUrlTransformed: 'dog.jpg',
    },
  },
  donations: [
    {
      total: 50,
      dateCreated: '2022-09-25T16:00:00.000Z',
      user: {
        name: 'Colin',
        avatar: {
          publicUrlTransformed: 'dog.jpg',
        },
      },
    },
  ],
});

const fakeUser = (overrides) => ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  createdOn: '2022-05-25T16:00:00.000Z',
  avatar: {
    publicUrlTransformed: 'colin.jpg',
  },
  fundraisers: [
    {
      id: '123',
      name: 'fundraiser1',
      description: 'fundraiser',
      amount: 5000,
      goal: 6000,
      dateCreated: '2022-09-25T16:00:00.000Z',
      photo: {
        image: {
          publicUrlTransformed: 'fundraiser1.jpg',
        },
      },
    },
  ],
  donations: [
    {
      id: '123',
      total: 25,
      dateCreated: '2022-05-26T16:00:00.000Z',
      fundraiser: {
        id: '456',
        name: 'fundraiser2',
        description: 'fundraiser',
        photo: {
          image: {
            publicUrlTransformed: 'fundraiser2.jpg',
          },
        },
      },
    },
  ],
  permissions: ['OVERLORD'],

  ...overrides,
});

const fakeOrderItem = () => ({
  __typename: 'OrderItem',
  id: casual.uuid,
  image: {
    image: `${casual.word}.jpg`,
  },
  name: casual.words(),
  price: 4234,
  quantity: 1,
  description: casual.words(),
});

const fakeOrder = () => ({
  __typename: 'Order',
  id: 'ord123',
  charge: 'ch_123',
  total: 40000,
  items: [fakeOrderItem(), fakeOrderItem()],
  createdAt: '2022-12-11T20:16:13.797Z',
  user: fakeUser(),
});

const fakeCartItem = (overrides) => ({
  __typename: 'CartItem',
  id: 'omg123',
  quantity: 3,
  product: fakeItem(),
  user: fakeUser(),
  ...overrides,
});


class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

function makePaginationMocksFor(length) {
  return [
    {
      request: { query: PAGINATION_COUNT_QUERY },
      result: {
        data: {
          _allFundraisersMeta: {
            count: length,
          },
          itemsConnection: {
            __typename: 'aggregate',
            aggregate: {
              count: length,
              __typename: 'count',
            },
          },
        },
      },
    },
  ];
}

export {
  makePaginationMocksFor,
  LocalStorageMock,
  fakeFundraiser,
  fakeUser,
  fakeCartItem,
  fakeOrder,
  fakeOrderItem,
};
