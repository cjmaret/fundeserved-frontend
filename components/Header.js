import Link from 'next/link';
import {
  HeaderComponent,
  Logo,
  NavLink,
  NavButton,
} from './styles/styledHeader';
import { useContext, useEffect, useState } from 'react';
import HeaderLogoImage from '../images/header-logo-2.png';
import MenuWrapper from './MenuWrapper';
import { MobileContext } from '../contexts/mobileContext';
import { useUser } from './User';
import SignOut from './SignOut';
import Search from './Search';

export default function Header() {
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const [isSearchMenuOpen, setIsSearchMenuOpen] = useState(false);
  const mobileMenu = useContext(MobileContext);
  const user = useUser();

  function closeAllMenus() {
    setIsMenuOpen(false);
    setIsSearchMenuOpen(false);
  }

  // one menu at a time
  useEffect(() => {
    function toggleMenus(e) {
      if (e.target.name === 'mobile-menu') {
        setIsSearchMenuOpen(false);
      } else if (e.target.name === 'mobile-search') {
        setIsMenuOpen(false);
      }
    }
    window.addEventListener('click', toggleMenus);
    return () => window.removeEventListener('click', toggleMenus);
  }, [isMenuOpen, isSearchMenuOpen]);

  function renderLogo() {
    return (
      <Logo mobileMenu={mobileMenu}>
        <Link href="/">
          <NavLink className="logo__anchor">
            <img
              className="logo-image"
              src={HeaderLogoImage}
              onClick={closeAllMenus}
              alt=""
            />
          </NavLink>
        </Link>
      </Logo>
    );
  }

  return (
    <HeaderComponent mobileMenu={mobileMenu}>
      <Search
        isSearchMenuOpen={isSearchMenuOpen}
        setIsSearchMenuOpen={setIsSearchMenuOpen}
        closeAllMenus={closeAllMenus}
        mobileMenu={mobileMenu}
      />
      {mobileMenu && renderLogo()}
      <MenuWrapper
        isMenuOpen={isMenuOpen}
        setIsMenuOpen={setIsMenuOpen}
        mobileMenu={mobileMenu}>
        <Link href="/fundraisers">
          <NavLink onClick={closeAllMenus}>All Fundraisers</NavLink>
        </Link>
        <Link href="/learn-more">
          <NavLink onClick={closeAllMenus}>How it works</NavLink>
        </Link>
        {!mobileMenu && renderLogo()}
        {user && (
          <>
            <Link href="/profile">
              <NavLink onClick={closeAllMenus}>My Profile</NavLink>
            </Link>
            <SignOut />
          </>
        )}
        {!user && (
          <>
            <Link href="/sign-in">
              <NavLink onClick={closeAllMenus}>Sign In</NavLink>
            </Link>
            <Link href="/sign-up">
              <NavLink onClick={closeAllMenus}>Sign Up</NavLink>
            </Link>
          </>
        )}
        <Link href="/create-fundraiser">
          <NavButton onClick={closeAllMenus} mobileMenu={mobileMenu}>
            Start a Fundraiser
          </NavButton>
        </Link>
      </MenuWrapper>
    </HeaderComponent>
  );
}
