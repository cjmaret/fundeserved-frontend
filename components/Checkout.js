import { gql, useMutation } from '@apollo/client';
import {
  CardElement,
  Elements,
  useElements,
  useStripe,
} from '@stripe/react-stripe-js';
import { loadStripe } from '@stripe/stripe-js';
import { useRouter } from 'next/dist/client/router';
import nProgress from 'nprogress';
import { useState } from 'react';
import styled from 'styled-components';
import { formatDollarsToCents } from '../lib/formatMoney';
import { CURRENT_USER_QUERY } from './User';

const CheckoutFormComponent = styled.form`
  display: grid;
  grid-gap: 1rem;
  padding-bottom: 3rem;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  & > button {
    background-color: ${(props) => (props.loading ? 'grey' : 'var(--red)')};
    color: white;
    font-size: 1.6rem;
    width: 125px;
    margin: 20px auto 0 auto;
    border: none;
    border-radius: 5px;
    padding: 6px 10px;
    transition: opacity 0.4s ease;
    &:hover {
      cursor: pointer;
      opacity: 0.7;
    }
  }
  .card {
    border: 1px solid grey;
    border-radius: 5px;
    padding: 10px 10px;
  }
`;

const CREATE_ORDER_MUTATION = gql`
  mutation CREATE_ORDER_MUTATION(
    $token: String!
    $amount: Int!
    $fundraiserId: String!
  ) {
    checkout(token: $token, amount: $amount, fundraiserId: $fundraiserId) {
      id
    }
  }
`;

const stripeLib = loadStripe(process.env.NEXT_PUBLIC_STRIPE_KEY);

function CheckoutForm({ amount, fundraiserId }) {
  const [error, setError] = useState();
  const [loading, setLoading] = useState(false);
  const stripe = useStripe();
  const elements = useElements();
  const router = useRouter();

  const [checkout, { error: graphQLError }] = useMutation(
    CREATE_ORDER_MUTATION,
    {
      refetchQueries: [{ query: CURRENT_USER_QUERY }],
    }
  );

  async function handleSubmit(e) {
    // stop form from submitting and turn loader on
    e.preventDefault();
    setLoading(true);
    // 2. start the page transition
    nProgress.start();
    // 3. create the payment method via stripe (token comes back here if successful)
    const { error, paymentMethod } = await stripe.createPaymentMethod({
      type: 'card',
      card: elements.getElement(CardElement),
    });
    // 4. handle any errors from stripe
    if (error) {
      setError(error);
      nProgress.done();
      return; // stops the checkout from happening
    }
    // 5. send the token from step 3 to our keystone server, via a custom mutation
    await checkout({
      variables: {
        token: paymentMethod.id,
        amount: formatDollarsToCents(amount),
        fundraiserId,
      },
    })
      .then((res) => {
        router.push({
          pathname: `/my-fundraisers`,
        });
      })
      .catch((err) => console.error(err));
    // 6. change the page to view the order
    // 8. turn off loader
    setLoading(false);
    nProgress.done();
  }
  return (
    <CheckoutFormComponent onSubmit={handleSubmit}>
      {graphQLError && <p>{graphQLError.message}</p>}
      <CardElement className="card" />
      <button disabled={loading} loading={loading}>
        {loading ? 'One moment' : 'Donate now'}
      </button>
    </CheckoutFormComponent>
  );
}

// useStripe needs to be wrapped in the Elements provider before it can be used
export default function Checkout({ amount, fundraiserId }) {
  return (
    <Elements stripe={stripeLib}>
      <CheckoutForm amount={amount} fundraiserId={fundraiserId} />
    </Elements>
  );
}
