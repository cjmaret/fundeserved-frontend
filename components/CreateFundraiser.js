import { useMutation } from '@apollo/client';
import gql from 'graphql-tag';
import Router from 'next/router';
import { formatDollarsToCents } from '../lib/formatMoney';
import useForm from '../lib/useForm';
import DisplayError from './ErrorMessage';
import { Form } from './styles/styledForm';
import { ALL_FUNDRAISERS_QUERY } from './Fundraisers';
import { CURRENT_USER_QUERY } from './User';
import { PAGINATION_COUNT_QUERY } from './Pagination';
import {
  CreateFundraiserComponent,
  FormGroup,
  FormSection,
  InstructionsSection,
  InstructionsSubtitle,
  InstructionsTitle,
} from './styles/styledCreateFundraiser';
import { MobileContext } from '../contexts/mobileContext';
import { useContext } from 'react';

export const CREATE_FUNDRAISER_MUTATION = gql`
  mutation CREATE_FUNDRAISER_MUTATION(
    $name: String!
    $description: String!
    $goal: Int!
    $image: Upload
  ) {
    createFundraiser(
      data: {
        name: $name
        description: $description
        amount: 0
        goal: $goal
        status: "ACTIVE"
        photo: { create: { image: $image, altText: $name } }
      }
    ) {
      id
      goal
      description
      name
    }
  }
`;

export default function CreateFundraiser() {
  const mobileWidth = useContext(MobileContext);
  const { inputs, handleChange, resetForm, clearForm } = useForm({
    image: '',
    name: '',
    description: '',
    goal: '',
  });

  const [createFundraiser, { loading, error, data }] = useMutation(
    CREATE_FUNDRAISER_MUTATION,
    {
      variables: { ...inputs, goal: formatDollarsToCents(inputs.goal) },
      refetchQueries: [
        { query: ALL_FUNDRAISERS_QUERY },
        { query: CURRENT_USER_QUERY },
        { query: PAGINATION_COUNT_QUERY },
      ],
    }
  );

  const textAreaRows = mobileWidth ? 6 : 9;

  return (
    <CreateFundraiserComponent>
      <FormGroup>
        <InstructionsSection>
          <InstructionsTitle>
            Create your Fundeserved fundraiser here
          </InstructionsTitle>
          <InstructionsSubtitle>
            Tell us why your fundraiser won't help anyone
          </InstructionsSubtitle>
        </InstructionsSection>
        <FormSection>
          <Form
            onSubmit={async (e) => {
              e.preventDefault();
              await createFundraiser()
                .then((res) => {
                  clearForm();
                  Router.push({
                    pathname: `/fundraiser/${res.data.createFundraiser.id}`,
                  });
                })
                .catch((err) => console.error(err));
            }}>
            <DisplayError error={error} />
            <fieldset disabled={loading} aria-busy={loading}>
              <label htmlFor="name">
                Fundraiser Name
                <input
                  type="text"
                  id="name"
                  name="name"
                  placeholder="Name"
                  value={inputs.name}
                  onChange={handleChange}
                />
              </label>
              <label htmlFor="image">
                Fundraiser Image
                <input
                  required
                  type="file"
                  id="image"
                  name="image"
                  onChange={handleChange}
                />
              </label>
              <label htmlFor="description">
                Fundraiser Description
                <textarea
                  type="textarea"
                  id="description"
                  name="description"
                  placeholder="Description"
                  value={inputs.description}
                  onChange={handleChange}
                  rows={textAreaRows}
                />
              </label>
              <label htmlFor="goal">
                Fundraiser Goal
                <input
                  type="number"
                  id="goal"
                  name="goal"
                  placeholder="Goal"
                  value={inputs.goal}
                  onChange={handleChange}
                />
              </label>
              <div className="button-group">
                <button type="submit">Create Fundraiser</button>
                <button type="button" onClick={clearForm}>
                  Clear Form
                </button>
              </div>
            </fieldset>
          </Form>
        </FormSection>
      </FormGroup>
    </CreateFundraiserComponent>
  );
}
