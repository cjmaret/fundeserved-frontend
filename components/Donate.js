import gql from 'graphql-tag';
import { useQuery } from '@apollo/client';
import { useEffect, useState } from 'react';
import { useContext } from 'react';
import { MobileContext } from '../contexts/mobileContext';

import Checkout from '../components/Checkout';
import {
  Cents,
  DetailsSubtitle,
  DetailsTitle,
  Dollar,
  DonateGroup,
  DonateInput,
  DonateInputError,
  DonateInputGroup,
  DonateSection,
  DonateTitle,
  GuaranteeDetailsGroup,
  GuaranteeGroup,
  GuaranteeImage,
  GuaranteeSubtitle,
  GuaranteeTitle,
  InfoDetailsGroup,
  InfoGroup,
  InfoImage,
  MainGroup,
  ReturnButton,
  TallySidebarGroup,
  TallyAmount,
  TallyDue,
  TallyTitle,
  TallyMobileGroup,
} from '../components/styles/styledDonate';
import DisplayError from './ErrorMessage';
import GuaranteeImageIcon from '../images/guarantee.png';

const SINGLE_FUNDRAISER_QUERY = gql`
  query SINGLE_FUNDRAISER_QUERY($id: ID!) {
    Fundraiser(where: { id: $id }) {
      id
      name
      description
      status
      amount
      goal
      photo {
        id
        altText
        image {
          publicUrlTransformed
        }
      }
    }
  }
`;

export default function Donate({ id }) {
  const mobileWidth = useContext(MobileContext);
  const { data, loading, error } = useQuery(SINGLE_FUNDRAISER_QUERY, {
    variables: {
      id,
    },
  });

  const Fundraiser = data?.Fundraiser || null;
  const [amountValue, setAmountValue] = useState('');
  const [inputError, setInputError] = useState(false);

  function handleAmountChange(e) {
    let amount = e.target.value.replace(/[^0-9.]/g, '');
    // let limitedAmount = Math.max(0, Math.min(50000, amount)).toString();
    setAmountValue(amount);
  }

  useEffect(() => {
    if (amountValue > 50000) {
      setInputError(true);
    } else {
      setInputError(false);
    }
  }, [amountValue]);

  function addCommasToNumber(num) {
    return num.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
  }

  return (
    <DonateSection>
      <MainGroup mobileWidth={mobileWidth}>
        <ReturnButton href={`/fundraiser/${id}`}>
          Return to fundraiser
        </ReturnButton>
        <InfoGroup>
          <InfoImage
            src={Fundraiser?.photo?.image.publicUrlTransformed}
            alt={Fundraiser?.name}
          />
          <DetailsTitle>
            You're supporting <span>{Fundraiser?.name}</span>
          </DetailsTitle>
        </InfoGroup>
        <DonateGroup>
          <DonateTitle>Enter your donation</DonateTitle>
          <DonateInputGroup inputError={inputError}>
            <Dollar>$</Dollar>
            <DonateInput
              type="text"
              onChange={handleAmountChange}
              value={amountValue}
              maxLength="5"
            />
            <Cents>.00</Cents>
          </DonateInputGroup>
          {inputError && (
            <DonateInputError>Donation limit is $50,000.00</DonateInputError>
          )}
        </DonateGroup>
        {mobileWidth && (
          <TallyMobileGroup>
            <TallyTitle>Your donation</TallyTitle>
            <TallyAmount>
              <span>Your donation</span>
              <span>${addCommasToNumber(amountValue)}.00</span>
            </TallyAmount>
            <TallyDue>
              <span>Total due today</span>
              <span>${addCommasToNumber(amountValue)}.00</span>
            </TallyDue>
          </TallyMobileGroup>
        )}
        <Checkout amount={amountValue} fundraiserId={Fundraiser?.id} />
        <GuaranteeGroup>
          <GuaranteeImage src={GuaranteeImageIcon} alt="guarantee icon" />
          <GuaranteeDetailsGroup>
            <GuaranteeTitle>Fundeserved donation protection</GuaranteeTitle>
            <GuaranteeSubtitle>
              Fundeserved guarantees 100% of your donation will go to whatever
              legal/ethical misuse this fundraiser claims to be perpetrating.
              However, if any of our staff members are skimming or otherwise
              embezzling donations to satisfy their own personal greed, then of
              course your donation is not protected by this guarantee.
            </GuaranteeSubtitle>
          </GuaranteeDetailsGroup>
        </GuaranteeGroup>
      </MainGroup>
      {!mobileWidth && (
        <TallySidebarGroup>
          <TallyTitle>Your donation</TallyTitle>
          <TallyAmount>
            <span>Your donation</span>
            <span>${addCommasToNumber(amountValue)}.00</span>
          </TallyAmount>
          <TallyDue>
            <span>Total due today</span>
            <span>${addCommasToNumber(amountValue)}.00</span>
          </TallyDue>
        </TallySidebarGroup>
      )}
    </DonateSection>
  );
}
