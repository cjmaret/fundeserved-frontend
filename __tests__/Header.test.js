import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import Header from '../components/Header';
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser } from '../lib/testUtils';

const notSignedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { authenticatedItem: null } },
  },
];

const signedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { authenticatedItem: fakeUser() } },
  },
];

describe('<Header />', () => {
  it('renders a minimal header when signed out', () => {
    const { container, debug } = render(
      <MockedProvider mocks={notSignedInMocks}>
        <Header />
      </MockedProvider>
    );
    expect(container).toHaveTextContent('Sign In');
    expect(container).toMatchSnapshot();
    const link = screen.getByText('Sign In');
    expect(link).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders a full header when signed in', async () => {
    const { container, debug } = render(
      <MockedProvider mocks={signedInMocks}>
        <Header />
      </MockedProvider>
    );
    // wait until rerender with fetched data
    await screen.findByText('My Profile');
    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent('Sign Out');
  });
});
