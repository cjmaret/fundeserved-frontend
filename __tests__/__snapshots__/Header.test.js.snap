// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders a full header when signed in 1`] = `
<div>
  <header
    class="styledHeader__HeaderComponent-sc-zui0ns-0 hEvuag"
  >
    <div
      class="styledSearch__SearchHeaderGroup-sc-q3k94-0 hcocDp"
    >
      <div
        class="styledHeader__IconWrapper-sc-zui0ns-5 styledSearch__SearchIconWrapper-sc-q3k94-1 dcDjBj"
      >
        <img
          alt=""
          class="icon-image"
          name="mobile-search"
          src="[object Object]"
        />
      </div>
    </div>
    <div
      class="styledHeader__MobileDropdown-sc-zui0ns-6 styledSearch__SearchMenu-sc-q3k94-2 hHOOVM lptXsH"
    >
      <div
        class="styledSearch__SearchComponent-sc-q3k94-3 jquTMK"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="downshift-0-menu"
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            aria-controls="downshift-0-menu"
            aria-labelledby="downshift-0-label"
            autocomplete="off"
            class=""
            id="search"
            placeholder="Search for a fundraiser"
            type="search"
            value=""
          />
        </div>
        <div
          aria-labelledby="downshift-0-label"
          class="styledSearch__DropDown-sc-q3k94-4 fvMoFz"
          id="downshift-0-menu"
          role="listbox"
        />
      </div>
    </div>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      Fundraisers
    </a>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      How it works
    </a>
    <div
      class="styledHeader__Logo-sc-zui0ns-1 LsOqq"
    >
      <a
        class="styledHeader__NavLink-sc-zui0ns-2 govIsg logo__anchor"
      >
        <img
          alt=""
          class="logo-image"
          src="[object Object]"
        />
      </a>
    </div>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      My Profile
    </a>
    <button
      class="SignOut__SignOutButton-sc-oloevi-0 cMdazv"
      type="button"
    >
      Sign Out
    </button>
    <a
      class="styledHeader__NavButton-sc-zui0ns-3 eEHdYO"
    >
      Start a Fundraiser
    </a>
  </header>
</div>
`;

exports[`<Header /> renders a minimal header when signed out 1`] = `
<div>
  <header
    class="styledHeader__HeaderComponent-sc-zui0ns-0 hEvuag"
  >
    <div
      class="styledSearch__SearchHeaderGroup-sc-q3k94-0 hcocDp"
    >
      <div
        class="styledHeader__IconWrapper-sc-zui0ns-5 styledSearch__SearchIconWrapper-sc-q3k94-1 dcDjBj"
      >
        <img
          alt=""
          class="icon-image"
          name="mobile-search"
          src="[object Object]"
        />
      </div>
    </div>
    <div
      class="styledHeader__MobileDropdown-sc-zui0ns-6 styledSearch__SearchMenu-sc-q3k94-2 hHOOVM lptXsH"
    >
      <div
        class="styledSearch__SearchComponent-sc-q3k94-3 jquTMK"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="downshift-0-menu"
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            aria-controls="downshift-0-menu"
            aria-labelledby="downshift-0-label"
            autocomplete="off"
            class=""
            id="search"
            placeholder="Search for a fundraiser"
            type="search"
            value=""
          />
        </div>
        <div
          aria-labelledby="downshift-0-label"
          class="styledSearch__DropDown-sc-q3k94-4 fvMoFz"
          id="downshift-0-menu"
          role="listbox"
        />
      </div>
    </div>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      Fundraisers
    </a>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      How it works
    </a>
    <div
      class="styledHeader__Logo-sc-zui0ns-1 LsOqq"
    >
      <a
        class="styledHeader__NavLink-sc-zui0ns-2 govIsg logo__anchor"
      >
        <img
          alt=""
          class="logo-image"
          src="[object Object]"
        />
      </a>
    </div>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      Sign In
    </a>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      Sign Up
    </a>
    <a
      class="styledHeader__NavButton-sc-zui0ns-3 eEHdYO"
    >
      Start a Fundraiser
    </a>
  </header>
</div>
`;

exports[`<Header /> renders a minimal header when signed out 2`] = `
<div>
  <header
    class="styledHeader__HeaderComponent-sc-zui0ns-0 hEvuag"
  >
    <div
      class="styledSearch__SearchHeaderGroup-sc-q3k94-0 hcocDp"
    >
      <div
        class="styledHeader__IconWrapper-sc-zui0ns-5 styledSearch__SearchIconWrapper-sc-q3k94-1 dcDjBj"
      >
        <img
          alt=""
          class="icon-image"
          name="mobile-search"
          src="[object Object]"
        />
      </div>
    </div>
    <div
      class="styledHeader__MobileDropdown-sc-zui0ns-6 styledSearch__SearchMenu-sc-q3k94-2 hHOOVM lptXsH"
    >
      <div
        class="styledSearch__SearchComponent-sc-q3k94-3 jquTMK"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="downshift-0-menu"
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            aria-controls="downshift-0-menu"
            aria-labelledby="downshift-0-label"
            autocomplete="off"
            class=""
            id="search"
            placeholder="Search for a fundraiser"
            type="search"
            value=""
          />
        </div>
        <div
          aria-labelledby="downshift-0-label"
          class="styledSearch__DropDown-sc-q3k94-4 fvMoFz"
          id="downshift-0-menu"
          role="listbox"
        />
      </div>
    </div>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      Fundraisers
    </a>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      How it works
    </a>
    <div
      class="styledHeader__Logo-sc-zui0ns-1 LsOqq"
    >
      <a
        class="styledHeader__NavLink-sc-zui0ns-2 govIsg logo__anchor"
      >
        <img
          alt=""
          class="logo-image"
          src="[object Object]"
        />
      </a>
    </div>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      Sign In
    </a>
    <a
      class="styledHeader__NavLink-sc-zui0ns-2 govIsg"
    >
      Sign Up
    </a>
    <a
      class="styledHeader__NavButton-sc-zui0ns-3 eEHdYO"
    >
      Start a Fundraiser
    </a>
  </header>
</div>
`;
